lib/Logger/LogThread.cpp

```diff
@@ -26,6 +26,12 @@
 #include "Logger/LogAppender.h"
 #include "Logger/Logger.h"
 
+// Hard-coded limit of delayed messages. If this limit is reached, we will
+// directly log all following messages. This is to prevent the queue from
+// growing indefinitely if the log thread is not able to keep up with the
+// incoming messages.
+static constexpr size_t MAX_QUEUED_MESSAGES = 10000;
```

Comment:
It would be really nice if this limit were configurable. LoggerFeature has a lot of options already, so we could easily add this one too.